File: src/python/core/src/dicebear/utils/xml.py
"""Minimal XML escaping helper for SVG/XML text and attribute content."""

from __future__ import annotations

_ENTITIES = {
    "&": "&amp;",
    "'": "&apos;",
    '"': "&quot;",
    "<": "&lt;",
    ">": "&gt;",
}

_TABLE = str.maketrans(_ENTITIES)


class Xml:
    """Minimal XML escaping helper for SVG/XML text and attribute content."""

    @staticmethod
    def escape(value: str) -> str:
        """Return ``value`` with the five XML predefined entities escaped."""
        return value.translate(_TABLE)

File: src/csharp/core/src/Utils/Xml.cs
using System.Text;

namespace DiceBear.Internal
{
    /// <summary>
    /// Minimal XML escaping helper for SVG/XML text and attribute content.
    /// </summary>
    internal static class Xml
    {
        /// <summary>
        /// Returns <paramref name="value"/> with the five XML predefined
        /// entities escaped. The substitution walks the string once, so an
        /// escaped '&amp;' is never escaped again.
        /// </summary>
        internal static string Escape(string value)
        {
            var builder = new StringBuilder(value.Length);

            foreach (var ch in value)
            {
                switch (ch)
                {
                    case '&':
                        builder.Append("&amp;");
                        break;
                    case '\'':
                        builder.Append("&apos;");
                        break;
                    case '"':
                        builder.Append("&quot;");
                        break;
                    case '<':
                        builder.Append("&lt;");
                        break;
                    case '>':
                        builder.Append("&gt;");
                        break;
                    default:
                        builder.Append(ch);
                        break;
                }
            }

            return builder.ToString();
        }
    }
}
